File: reverse-integer/ReverseInteger.java
class Solution {
    public int reverse(int x) {
        int sum = 0, digit = 0, prev = 0;
        while(x != 0){
            digit = x % 10;
            sum = sum*10 + digit; 
            x /= 10;
            
            if((sum - digit) / 10 != prev )
                return 0;
            
            prev = sum;
        }

        return sum;
    }
}

File: reverse-integer/ReverseInteger.kt
class Solution {
    fun reverse(x: Int): Int {
        var temp = x
        var reversed =0
        var prev  = 0

        while(temp !=0){
            var remainder = temp % 10
            reversed = reversed * 10 + remainder 

            if((reversed - remainder) / 10 != prev ) return 0
            temp /= 10
            prev = reversed
        }
        return reversed
    }
}